src/App.tsx

```diff
@@ -16,80 +17,132 @@ export const goodsFromServer = [
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
-  { sortType, isReversed }: ReorderOptions,
+  { isReversed, sortType }: State,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+export class App extends React.Component {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-info',
+              {
+                'is-light': this.state.sortType !== SortType.ALPHABET,
+              },
+            )}
+            onClick={() => {
+              this.setState(
+                { sortType: SortType.ALPHABET },
+              );
+            }}
+          >
+            Sort alphabetically
+          </button>
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              {
+                'is-light': this.state.sortType !== SortType.LENGTH,
+              },
+            )}
+            onClick={() => {
+              this.setState(
+                { sortType: SortType.LENGTH },
+              );
+            }}
+          >
+            Sort by length
+          </button>
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-warning',
+              {
+                'is-light': !this.state.isReversed,
+              },
+            )}
+            onClick={() => {
+              this.setState((prevState: State) => (
+                { isReversed: !prevState.isReversed }
+              ));
+            }}
```

Comment:
it is bad practice to make anonymous arrow function in JSX
better to make handler function and pass reference here
```
```suggestion
            onClick={onReset}
```
  const onReset = () => {
     this.setState((prevState: State) => (
        { isReversed: !prevState.isReversed }
     ));
  }